ui/js/pods-dfv/_src/pick/views/select-view.js

```diff
@@ -145,19 +145,57 @@ export const SelectView = Marionette.CollectionView.extend( {
 
 	/**
 	 * Setup to be done once attached to the DOM.  Select2 has some setup needs.
+	 *
+	 * @var {RelationshipCollection} this.collection
 	 */
 	onAttach: function () {
+		const view_name = this.fieldConfig.view_name;
+		const format_type = this.fieldConfig.pick_format_type;
+		const limit = this.fieldConfig.pick_limit;
+		const numSelected = this.collection.filterBySelected().length;
 
-		if ( this.fieldConfig.view_name === 'select2' ) {
+		// Initialize select2 fields
+		if ( 'select2' === view_name ) {
 			this.setupSelect2();
 		}
+
+		// Check initial selection limit status for regular multiselect and enforce it if needed
+		if ( 'select' === view_name && 'multi' === format_type && numSelected >= limit ) {
+			this.disableUnselected();
+		}
 	},
 
 	/**
-	 *
+	 * @var {RelationshipCollection} this.collection
 	 */
 	onChangeSelected: function () {
+		const view_name = this.fieldConfig.view_name;
+		const format_type = this.fieldConfig.pick_format_type;
+		const limit = this.fieldConfig.pick_limit;
+		let numSelected;
+
 		this.collection.setSelected( this.$el.val() );
+		numSelected = this.collection.filterBySelected().length;
+
+		if ( 'select' === view_name && 'multi' === format_type ) {
+			if ( numSelected >= limit ) {
```

Comment:
This should check that limit itself is `0 < limit` and please follow the `1 < 2` order of vars when comparing. So it'd be `limit <= numSelected`.

If limit is not above 0, then it shouldn't run enableAll() either.